lib/features/modeling/cmd/SpaceToolHandler.js

```diff
@@ -81,7 +82,7 @@ SpaceToolHandler.prototype.updateConnectionWaypoints = function(
   forEach(connections, function(connection) {
     var source = connection.source,
         target = connection.target,
-        waypoints = connection.waypoints;
+        waypoints = copyWaypoints(connection);
```

Comment:
Why is this necessary?

Reply:
Mutation breaks undo/redo. By the way, this PR is work in progress.